From 34e4e19842711f625d82df8477305869339b96e3 Mon Sep 17 00:00:00 2001
From: Vasantha Balla <vballa@codeaurora.org>
Date: Mon, 16 Oct 2017 17:23:54 +0530
Subject: [PATCH 8/9] libmedia: Fix null pointer crash in secure buffer
 allocation..

Add null pointer check for native_handle as it is null if
secure memory allocation fails.

Change-Id: I204ca69b9808448788c83a1ae5a642618eca8973
---
 media/libmedia/omx/1.0/WOmxNode.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/media/libmedia/omx/1.0/WOmxNode.cpp b/media/libmedia/omx/1.0/WOmxNode.cpp
index 0b40e8d..adb2f84 100644
--- a/media/libmedia/omx/1.0/WOmxNode.cpp
+++ b/media/libmedia/omx/1.0/WOmxNode.cpp
@@ -151,8 +151,9 @@ status_t LWOmxNode::allocateSecureBuffer(
                     hidl_handle const& outNativeHandle) {
                 fnStatus = toStatusT(status);
                 *buffer = outBuffer;
-                *native_handle = NativeHandle::create(
-                        native_handle_clone(outNativeHandle), true);
+                *native_handle = outNativeHandle == nullptr ?
+                               nullptr : NativeHandle::create(
+                               native_handle_clone(outNativeHandle), true);
             }));
     return transStatus == NO_ERROR ? fnStatus : transStatus;
 }
-- 
2.7.4

